### PR TITLE
Hw-event-proxy cluster deployment

### PR DESCRIPTION
**Pull request by jzding**
Added cluster deployment yaml files.
Added Makefile for cluser deployment.
Added env MSG_PARSER_TIMEOUT for customizing message parser timeout.
Added secret for Redfish basic auth.

Signed-off-by: Jack Ding <jacding@redhat.com>

### CODE REVIEW

**Review by nishant-parekh — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,47 @@
+.PHONY: build
+
+# Current  version
+VERSION ?=latest
+
+# Default image tag
+PROXY_IMG ?= quay.io/redhat_emp1/hw-event-proxy:$(VERSION)
+SIDECAR_IMG ?= quay.io/redhat_emp1/cloud-event-proxy:$(VERSION)
```
Comment (aneeshkp) [inline Makefile:8]: @nishant-parekh Do you thin we can replace this to official repo?

File: docs/development.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+## Running examples locally
+
+The Hw Event Proxy works with [Cloud Event Proxy](https://github.com/redhat-cne/cloud-event-proxy).
+Run cloud-event-proxy sidecar and consumer example from the cloud-event-proxy repo for testing locally.
+
+### Set environment variables
+```
+export NODE_NAME=mynode
```
Comment (aneeshkp) [inline docs/development.md:8]: Node name you can read from spec like this as part of deployment.yaml ?
 env:
            - name: NODE_NAME
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName

Comment (aneeshkp) [reply docs/development.md:8]: example: https://github.com/redhat-cne/cloud-event-proxy/blob/main/examples/consumer/deployment/deployment.yaml#L35

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+export PROXY_IMG=quay.io/redhat_emp1/hw-event-proxy:${VERSION}
+export SIDECAR_IMG=quay.io/redhat_emp1/cloud-event-proxy:${VERSION}
+export CONSUMER_IMG=quay.io/redhat_emp1/cloud-native-event-consumer:${VERSION}
+```
+
+### Setup AMQ Interconnect
+
+Install the `Red Hat Integration - AMQ Interconnect` operator in a new namespace `<AMQP_NAMESPAVCE>` namespace from the OpenShift Web Console.
+
+Open the `Red Hat Integration - AMQ Interconnect` operator, click `Create Interconnect` from the `Red Hat Integration - AMQ Interconnect` tab. Use default values and make sure the name is `amq-interconnect`.
```
Comment (aneeshkp) [inline docs/development.md:86]: You can deploy  interconect  via script if  it helps 
```


   
   
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: amq7-interconnect-operator
  namespace: amqp-interconnect
spec:
  targetNamespaces:
  - amqp-interconnect

oc get packagemanifest amq7-interconnect-operator -n openshift-marketplace -o jsonpath='{.status.defaultChannel}'
// replace the output as channel value 
//if 1.10.x doesn't work try 1.2.0

apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: amq7-interconnect-subscription
  namespace: amqp-interconnect
spec:
  channel: 1.10.x 
  name:  amq7-interconnect-operator
  source: redhat-operators 
  sourceNamespace: openshift-marketplace
  
  // After Successfully installed CSV create an instance
apiVersion: interconnectedcloud.github.io/v1alpha1
kind: Interconnect
metadata:
  name: amq-interconnect
  namespace: amqp-interconnect
spec:
  deploymentPlan:
    placement: Any
    role: interior
    size: 1
    ```

**Review by aneeshkp — APPROVED** [review]

File: docs/development.md (hunk 2 above)
Comment (jzding) [reply docs/development.md:8]: This section is for testing locally without cluster deployment.

**Pull request merged**